src/libs/actions/IOU.ts

```diff
@@ -1488,22 +1515,24 @@ function createSplitsAndOnyxData(participants, currentUserLogin, currentUserAcco
             const existingChatReport = ReportUtils.getChatByParticipants([accountID]);
             isNewOneOnOneChatReport = !existingChatReport;
             shouldCreateOptimisticPersonalDetails = isNewOneOnOneChatReport && !personalDetailExists;
-            oneOnOneChatReport = existingChatReport || ReportUtils.buildOptimisticChatReport([accountID]);
+            oneOnOneChatReport = existingChatReport ?? ReportUtils.buildOptimisticChatReport([accountID]);
         }
 
         // STEP 2: Get existing IOU/Expense report and update its total OR build a new optimistic one
         // For Control policy expense chats, if the report is already approved, create a new expense report
-        let oneOnOneIOUReport = oneOnOneChatReport.iouReportID ? lodashGet(allReports, `${ONYXKEYS.COLLECTION.REPORT}${oneOnOneChatReport.iouReportID}`, undefined) : undefined;
+        let oneOnOneIOUReport: OneOnOneIOUReport = oneOnOneChatReport.iouReportID ? allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${oneOnOneChatReport.iouReportID}`] ?? undefined : undefined;
         const shouldCreateNewOneOnOneIOUReport =
-            _.isUndefined(oneOnOneIOUReport) || (isOwnPolicyExpenseChat && ReportUtils.isControlPolicyExpenseReport(oneOnOneIOUReport) && ReportUtils.isReportApproved(oneOnOneIOUReport));
+            oneOnOneIOUReport === undefined || (isOwnPolicyExpenseChat && ReportUtils.isControlPolicyExpenseReport(oneOnOneIOUReport) && ReportUtils.isReportApproved(oneOnOneIOUReport));
 
-        if (shouldCreateNewOneOnOneIOUReport) {
+        if (oneOnOneIOUReport === undefined || shouldCreateNewOneOnOneIOUReport) {
             oneOnOneIOUReport = isOwnPolicyExpenseChat
-                ? ReportUtils.buildOptimisticExpenseReport(oneOnOneChatReport.reportID, oneOnOneChatReport.policyID, currentUserAccountID, splitAmount, currency)
+                ? ReportUtils.buildOptimisticExpenseReport(oneOnOneChatReport.reportID, oneOnOneChatReport.policyID ?? '', currentUserAccountID, splitAmount, currency)
                 : ReportUtils.buildOptimisticIOUReport(currentUserAccountID, accountID, splitAmount, oneOnOneChatReport.reportID, currency);
         } else if (isOwnPolicyExpenseChat) {
-            // Because of the Expense reports are stored as negative values, we subtract the total from the amount
-            oneOnOneIOUReport.total -= splitAmount;
+            if (typeof oneOnOneIOUReport?.total === 'number') {
```

Comment:
Why are we doing assertion here?

Reply:
Because `oneOnOneIOUReport?.total` can be `undefined`
![image](https://github.com/Expensify/App/assets/23176449/fe7a6bba-9f0e-43a9-9322-5cf8068eeafa)

And if I simplify the check to `if (typeof oneOnOneIOUReport?.total) {` it won't work for `0` value

Reply:
My concern is if `oneOnOneIOUReport` is defined and `total` is `null` for whatever reason (either because is wrongly typed or another situation), that would be possible to do the operation (although TS will complain anyway).

Assuming `oneOnOneIOUReport` is defined at this point:

- If `total` is `undefined`, the operation will result in a `NaN` value.
- If `total` is `null`, the operation will result in the correct value subtracted.

My suggestion would be to test this flow or, by safety, consider the situations above in the code.

Reply:
I've tested cases mentioned in this file and what I've found out:
- `total` field always was there, it had or negative number or `0` value
- I think we will need to update onyx `Report` type and define several types in a union depending on the Report `type` field, because it looks like `CONST.REPORT.TYPE.EXPENSE` type always have amount. But I don't think it's a good idea to do it in this PR, because it's a complex one and Report type is used in a lot of places.
- If we still worry about `null` case, we can check for `undefined`, something like
```
 if (oneOnOneIOUReport?.total !== undefined) {
```

@fabioh8010 What do you think?

Reply:
Hmm I see, maybe `total` is always defined then. Is possible to change `total` to be only `number`?

Reply:
@fabioh8010 Making `total` be required triggers TS errors, and as mentioned I guess it will be required only for some types of Reports (`CONST.REPORT.TYPE.EXPENSE` type for example)